reconcile/utils/vault.py

```diff
@@ -238,9 +238,19 @@ def write(self, secret, decode_base64=True):
             self._write_v1(secret_path, data)
 
     def _write_v2(self, path, data):
-        # do not forget to run `self._read_all_v2.cache_clear()`
-        # if this ever get's implemented
-        raise NotImplementedError('vault_client write v2')
+        path_split = path.split('/')
+        mount_point = path_split[0]
+        write_path = '/'.join(path_split[1:])
+
```

Comment:
I'm not very familiar with this class, but since reads in v2 are cached, I understand that the cache clear comments is pertinent and we would need to clear the lru cache in this method.

Reply:
added!